airflow/cli/commands/internal_api_command.py

```diff
@@ -222,7 +222,8 @@ def create_app(config=None, testing=False):
     if "SQLALCHEMY_ENGINE_OPTIONS" not in flask_app.config:
         flask_app.config["SQLALCHEMY_ENGINE_OPTIONS"] = settings.prepare_engine_args()
 
-    InternalApiConfig.force_database_direct_access("Gunicorn worker initialization")
+    if conf.getboolean("core", "database_access_isolation", fallback=False):
+        InternalApiConfig.set_use_database_access("Gunicorn worker initialization")
```

Comment:
Should we fail this command if `database_access_isolation` is `False`? The user is trying to run the internal API component but `database_access_isolation` is `False`

Reply:
Yes. Why not.  It would continue to work, byt it's more consistent this way.